src/components/SelectBox.js

```diff
@@ -29,80 +21,43 @@ const MarkAsDefault = withMsg(({ msg, value, isDefault }) => {
  *       __DropdownButton__ with a set of scrolling __MenuItem__s.
  */
 
-class SelectBox extends React.Component {
-  constructor (props) {
-    super(props)
-    const { sort, items, locale } = this.props
-    this.state = {
-      selected: getSelectedId(props),
-      items: getItems({ sort, items, locale }),
-    }
-    this.handleChange = this.handleChange.bind(this)
-    this.getValidationClass = this.getValidationClass.bind(this)
-  }
-
-  componentWillReceiveProps (nextProps) {
-    const { sort, items, locale } = nextProps
-    const nextState = { items: getItems({ sort, items, locale }) }
-    if (this.props.selected !== nextProps.selected) {
-      nextState.selected = getSelectedId(nextProps)
-    }
-    this.setState(nextState)
-  }
-
-  handleChange (id) {
-    return () => {
-      this.setState({ selected: id })
-      this.props.onChange(id)
-    }
-  }
-
-  getValidationClass () {
-    const { validationState } = this.props
-    switch (validationState) {
-      case 'error':
-        return style['selectBox-has-error']
-      default:
-        return ''
-    }
+const SelectBox = ({ sort, items = [], locale, selected, onChange, validationState, id, disabled }) => {
```

Comment:
The refactoring looks good.  Maybe mention in your PR description...

> Refactored SelectBox into **a stateless** functional component to remove deprecated lifecycle method use causing react warnings.

Reply:
It was extracted as a commit "SelectBox: remove componentWillReceiveProps "

Reply:
> It was extracted as a commit "SelectBox: remove componentWillReceiveProps "

Yeah, but you would have said it should be in a separate PR if I had posted this PR :wink: 

Really I just flipped to the Files changed tab and started looking at everything.  Had to go back and look at the commits etc to get an idea of the major things all going on at the same time (which needs to happen that way sometimes with large upgrades like this).

Reply:
@sjd78 @sgratch 
some commits (like "SelectBox..." or code formatting) could be extracted to separate PRs. However I would prefer to keep it all together because:
1. GH is not very good at tracking connected PRs (here it would be few of them)
2. without some fixes (i.e. after upgrading Immutables.js) the UI just won't start